src/Table/TableBody.js

```diff
@@ -129,14 +130,16 @@ class TableBody extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.allRowsSelected !== nextProps.allRowsSelected) {
-      if (!nextProps.allRowsSelected) {
-        this.setState({
-          selectedRows: [],
-        });
-      } else {
+    if (this.props.allRowsSelected && !nextProps.allRowsSelected) {
+      this.setState({
+        selectedRows: [],
+      });
+    } else {
+      const oldRows = this.calculatePreselectedRows(this.props);
```

Comment:
This change makes that block run after every render, even if the `allRowsSelected` property didn't change.
That's hiding something. I definitely think that the property fix isn't has that level of the code.